compiler/next/include/chpl/uast/New.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_NEW_H
+#define CHPL_UAST_NEW_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Expression.h"
+
+namespace chpl {
+namespace uast {
+
+/**
+  This class represents a new expression. For example:
+
+  \rst
+  .. code-block:: chapel
+
+    var foo = new bar(a = 1, 2);
+  \endrst
+
+  Creates a call expression where the base expression is 'new bar'.
+*/
+class New : public Expression {
+ public:
+
+  /**
+    Possible management flavors for a new expression.
+  */
+  enum struct Management {
```

Comment:
Everywhere else we are using unscoped enums (because they are "scoped" by the class they contain). I think this should follow, at least until we have a discussion about it. It seems to me that `New::Borrowed` is reasonably clear. Also we have ALLCAPS for enum values in the compiler.